Oop №1

LarionovKirill opened the pull request.

LarionovKirill committed:
Начало работы. Создание проекта.

LarionovKirill committed:
Feature(AirTravel.cs): Добавлены поля класса их свойства и конструкторы класса.
Feature (FlightType.cs): Создано перечисление типов полета.
Feaure (CollisionManager.cs): Создан класс проверки коллизий.

LarionovKirill committed:
feature(UserInterface.cs): Добавил пользовательский интерфейс и сверстал часть.

LarionovKirill committed:
feature(UserInterface.cs): добавлены методы и picturebox, но это еще не все

LarionovKirill committed:
Fix(UserInterface.cs): Исправлены методы обработки полей.

LarionovKirill committed:
Feature(UserInterface.cs): Добавлена проверка на длину полета и сделаны мелкие правки.

LarionovKirill committed:
Feature(UserInterface.cs): Добавил фото, но с методами доступа к ним пока еще туго.

LarionovKirill committed:
Feature(UserINterface.cs):Добавил изменение контраста кнопок при наведении на них.

LarionovKirill committed:
Feature(UserInterface.cs): Метод добавления теперь записывает значения в ListBox

LarionovKirill committed:
Feature(UserInterface.cs): Добавлен метод удаления

LarionovKirill committed:
Feature(MainForm.cs): Изменил название формы.

LarionovKirill committed:
Feature(UserInterface.cs): добавлены методы изменения значений перелета и метод сортировки данных.

LarionovKirill committed:
Feature(UserInterface.cs): Добавлена сериализация и десериализация.

LarionovKirill committed:
Style(UserInterface.cs): незначительные изменения.

LarionovKirill committed:
Fix(UserInterface.cs): Испарвил баг с неправильным отображением времени и городов

LarionovKirill committed:
Feature(ChangingForm.cs): Изменений много. Я создал дополнительную форму для того, чтобы через нее добавлять и изменять значения, а главная только для обзора данных.

LarionovKirill committed:
Fix(MainForm.cs): исправлено название лейбла.
Fix(UserInterface.cs): Исправлен баг с вызовом одного и того же участка кода.
Feature(UserInterface.cs): добалена обработка исключений.

LarionovKirill committed:
Я случайно создал папку в папке и было пустое вложение

LarionovKirill committed:
Fix(UserInterface.cs): исправил выравнивание элементов.
Fix(MainForm.cs): изменил максимальный и минимальный размер формы.

LarionovKirill committed:
Refactoring(PersonalTask.sln): расположил формы в папки

LarionovKirill committed:
Feature(project): добавил папку и первую лабу по ООП.

LarionovKirill committed:
Я тупанул, но вроде все нормально вернул.

LarionovKirill committed:
Добавление лабы№1 по ООП. (Я ее делал в другом проекте до индивидуальной, поэтому одним коммитом. )

LarionovKirill committed:
.

LarionovKirill committed:
Fix(OOP): Исправление недочетов в проекте.

LarionovKirill committed:
Fix(Uml лаба 1.jpg):

LarionovKirill committed:
Fix(Uml лаба 1.jpg): Исправил uml - диаграмму.

LarionovKirill committed:
Начало лабы №2.
Feature(Category.cs): добавлено перечисление.
Refactoring (Item.cs): теперь категориии являются композицией данного класса.
Feature (ItemsTab.cs): добавлена работа с перечислениями в клиентский код

LarionovKirill committed:
Feature(ItemsTabs.cs): Добавил обрабоку выбора категории и их изменение.

LarionovKirill committed:
Я опять не посмотрел в какой ветке работаю. Но я слил с нужной веткой и убрал все изменения. Благо их было немного.

LarionovKirill committed:
Fix(Project): везде уменьшил обращение к пространству имен.

LarionovKirill committed:
Fix(Customer.cs): исправена передача имени свойства в метод.

LarionovKirill committed:
.

LarionovKirill committed:
/

LarionovKirill committed:
Feature(gitignore)

LarionovKirill committed:
Delete slnx.sqlite

LarionovKirill committed:
Delete VSWorkspaceState.json

LarionovKirill committed:
Delete .suo

LarionovKirill committed:
Merge pull request #20 from LarionovKirill/tempOOP1

Temp oop1

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CustomerTab.cs:
```diff
@@ -0,0 +1,127 @@
+            }
+        }
+
+        private void FullNameTextBox_TextChanged(object sender, EventArgs e)
+        {
+            try
+            {
+                Services.ValueValidator.AssertStringOnLength(
+                    fullNameTextBox.Text,
+                    200,
+                    "FullNameTextBox");
+                fullNameTextBox.BackColor = Color.White;
+            }
+            catch
+            {
+                fullNameTextBox.BackColor = Color.Pink;
+            }
+        }
```
Не хватает логики по редактированию данных о товаре и о клиенте. Данные после валидации никуда не идут. Валидацию из View можно вообще убрать, так как она уже происходит в Model.

LarionovKirill replied:
Согласен, бред какой-то сделал, но уже исправил.

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Item.cs:
```diff
@@ -0,0 +1,121 @@
+	/// </summary>
+	public class Item
+	{
+		/// <summary>
+		/// Id товара.
+		/// </summary>
+		static private int _id;
```
Лишнее статическое закрытое поле, которое нигде не используется. Ниже у тебя есть **public int Id { get; private set; }** это авто свойство, при их использовании явно создавать закрытое поле не нужно, оно создаёт само. В **Customer.cs** такая же история.

LarionovKirill replied:
Вопрос про автосвойства. Предпочтение отдается им? Просто у меня немного как понимание с с++ и нужно сначала создать поле, а потом к нему геттер и сеттер, ну если так, то получается что почти весь класс должен состоять только из автосвойств?

Ershovoy replied:
Если в геттере или сеттере нету никакой логики, то лучше использовать автосвойство. Вдруг в будущем появится логика при обращении к полю или его присваивании, которую ты поместишь в геттер/сеттер свойства. В таком случае тебе не придётся во всем классе переименовывать _films в Films. И это ещё в лучшем случае, когда ты знаешь, что тебе нужно это сделать.

Ershovoy commented on ObjectOrientedPractics/OOP/View/MainForm.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace OOP
+{
+    public partial class MainForm : Form
```
Не хватает адаптивной верстки

LarionovKirill replied:
Блин, в первой забыл добавить(
Сделаю

Ershovoy replied:
Да уже увидел, что во второй есть. Тогда можешь не добавлять, наверное, тогда.

LarionovKirill replied:
Ну если учитывать ветку develop, то да, добавится уже там

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CustomerTab.cs:
```diff
@@ -0,0 +1,127 @@
+
+namespace OOP.View
+{
+    public partial class CustomerTab : UserControl
```
Не хватает XML комментариев для класса и методов.

LarionovKirill replied:
Сделал.

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Item.cs:
```diff
@@ -0,0 +1,121 @@
+		/// <summary>
+		/// Цена товара.
+		/// </summary>
+		private double _cost;
```
Для всего где требуется точные значение и ни в коем случае не допускаются погрешности, используется тип данных decimal. Double и float не точно представляют числа с плавающей точной (после определённого знака), но быстрее вычисляютсяя процессором. Может видел мемы где сравнивают два казалось бы одинаковых числа, но получают в результате false. Например 3 + 0.5 **не** равно 3.5, а равно что-то типо 3.50000134. Ходят легенды, что в ранней версии сбербанка была такого рода ошибка и некоторые успели ненадолго стать миллионерами.

LarionovKirill replied:
Decimal тип занимает больше памяти и поэтому погрешность меньше?

LarionovKirill committed:
Fix(Customer.cs):Убрал лишнее поле.
Fix(item.cs):Убрал лишнее поле и изменил тип данных стоимости.
Fix(ValueValidetor.cs):Изменил типы данных параметров метода.
Fix(CustomerTab.cs):Добавил XML, изменил методы с валидацией.
Fix(ItemsTab.cs):Изменил методы с валидацией и убрал лишний метод.
